rholang/src/main/scala/coop/rchain/rholang/interpreter/RholangCLI.scala

```diff
@@ -121,9 +123,9 @@ object RholangCLI {
     Console.println(PrettyPrinter().buildString(normalizedTerm))
   }
 
-  private def printStorageContents[F[_]](space: RhoISpace[F]): Unit = {
+  private def printStorageContents[F[_]: FlatMap](space: RhoISpace[F]): F[Unit] = {
     Console.println("\nStorage Contents:")
-    Console.println(StoragePrinter.prettyPrint(space))
+    StoragePrinter.prettyPrint(space).map(Console.println)
```

Comment:
No, it works

Reply:
Oh, since `Console.println` returns a `Unit` then you should closed it over `Sync.delay` because other wise it will print the moment you **construct** your `F`, not the moment the `F` is evaluated

```
Sync.delay(Console.println("\nStorage Contents:")) >>
    StoragePrinter.prettyPrint(space).map(Console.println)
```